src/canopy_alloc.F90

```diff
@@ -45,6 +45,10 @@ SUBROUTINE canopy_alloc
     if(.not.allocated(ppfd_ave))           allocate(ppfd_ave(modlays))
     if(.not.allocated(lad))                allocate(lad(nlat*nlon,modlays))
     if(.not.allocated(lad_3d))             allocate(lad_3d(nlon,nlat,modlays))
+    if(.not.allocated(zo_h))               allocate(zo_h(nlat*nlon))
+    if(.not.allocated(zoh_2d))             allocate(zoh_2d(nlon,nlat))
+    if(.not.allocated(d_h))                allocate(d_h(nlat*nlon))
+    if(.not.allocated(dh_2d))              allocate(dh_2d(nlon,nlat))
```

Comment:
It's a bit confusing that the names don't match between the 1-d and 2-d versions, with the 1-d using `_` before h but the 2-d not. Maybe pick one way to do it.